### PR TITLE
feat(a11y): Enable trapping focus only for Tab navigation

### PR DESCRIPTION
**Pull request by sdrozdsap**
Closes: https://jira.tools.sap/browse/CXSPA-7967

### CODE REVIEW

File: projects/core/src/features-config/feature-toggles/config/feature-toggles.ts [hunk 1]
```diff
@@ -308,6 +308,13 @@ export interface FeatureTogglesInterface {
    */
   a11yFocusableCarouselControls?: boolean;
 
+  /**
+   * Enables only Tab/Shift+Tab keyboard navigation in dialogs and preserved default scrolling behaviour of up/down keys.
+   * Components:
+   * - `PickupOptionDialogComponent`
+   */
```
Comment (sdrozdsap) [inline projects/core/src/features-config/feature-toggles/config/feature-toggles.ts:316]: Most probably more dialog components will have the same change thus the list format, for now with only one component though:)

File: feature-libs/pickup-in-store/components/container/pickup-option-dialog/pickup-option-dialog.component.ts [hunk 2]
```diff
@@ -47,19 +49,28 @@ export class PickupOptionDialogComponent implements OnInit, OnDestroy {
...
-    focusOnEscape: true,
-  };
-
   readonly ICON_TYPE = ICON_TYPE;
   /** The reason given closing the dialog window without selecting a location */
   readonly CLOSE_WITHOUT_SELECTION = 'CLOSE_WITHOUT_SELECTION';
   /** The reason given closing the dialog window after selecting a location */
   readonly LOCATION_SELECTED = 'LOCATION_SELECTED';
 
+  protected featureConfigService = inject(FeatureConfigService);
```
Comment (Zeyber) [inline feature-libs/pickup-in-store/components/container/pickup-option-dialog/pickup-option-dialog.component.ts]: We can make the featureConfigService `private` to make it easier to clean up. This is our "use of private" exception ;)

File: projects/storefrontlib/layout/a11y/keyboard-focus/keyboard-focus.model.ts [hunk 3]
```diff
@@ -109,10 +109,16 @@ export type TrapFocusType =
...
+ * Options:
+ * - Use the `trap` property to enable trapping for both Tab/Shift+Tab and Arrow Up/Down keys.
+ * - Use the `trapTab` property to trap only Tab/Shift+Tab, preserving the default behavior of the Arrow Up/Down keys.
  */
 export interface TrapFocusConfig extends TabFocusConfig {
-  /** traps the focus */
+  /** traps the focus for both tab/shift+tab and up/down keys */
   trap?: TrapFocusType;
+  /** traps the focus for tab/shift+tab */
+  trapTab?: TrapFocusType;
```
Comment (Zeyber) [inline projects/storefrontlib/layout/a11y/keyboard-focus/keyboard-focus.model.ts]: You've done a nice job of introducing this property elegantly while handling any breaking change. Thanks!

My only nitpick here is to name this `trapTabOnly` instead as it distinguishes it more. However, this is opinionated.

**Review by Zeyber — CHANGES_REQUESTED** [review]

File: feature-libs/pickup-in-store/components/container/pickup-option-dialog/pickup-option-dialog.component.ts (hunk 2 above)
Comment (sdrozdsap) [reply feature-libs/pickup-in-store/components/container/pickup-option-dialog/pickup-option-dialog.component.ts]: Sure!

File: projects/storefrontlib/layout/a11y/keyboard-focus/keyboard-focus.model.ts (hunk 3 above)
Comment (sdrozdsap) [reply projects/storefrontlib/layout/a11y/keyboard-focus/keyboard-focus.model.ts]: Thanks - I've changed to `trapTabOnly`

**Review by Zeyber — DISMISSED** [review]

**Review by developpeurweb — DISMISSED** [review]

**Review by Zeyber — APPROVED** [review]

**Pull request merged**